### PR TITLE
vagrant: 2.3.4 -> 2.3.7, vagrant-libvirt-provider: 0.8.2 -> 0.12.2

### PR DESCRIPTION
**Pull request by teutat3s**
###### Description of changes
- File in post fixup phase got deleted by upstream, no longer needed.
https://github.com/hashicorp/vagrant/commit/7db87b9da37e019e0201e0f8c7517da553a8910f

- Patch needed some changes to work with newest version of vagrant.
https://github.com/hashicorp/vagrant/commit/3422582d68e94bb1c4da64512ac9a2c3654237d6

https://github.com/hashicorp/vagrant/releases/tag/v2.3.5
https://github.com/hashicorp/vagrant/releases/tag/v2.3.6
https://github.com/hashicorp/vagrant/releases/tag/v2.3.7

https://github.com/vagrant-libvirt/vagrant-libvirt/compare/0.8.2...0.12.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nullie** [comment]
I'm already using this successfully with an overlay. Can I do something to help merge this?

**Comment by pbering** [comment]
> I'm already using this successfully with an overlay. Can I do something to help merge this?

Can you show your overlay? Would like to use this also 🙂

**Comment by nullie** [comment]
Sure 

    overlay-updates = final: prev: {
      # Vagrant 2.3.7
      vagrant =
        (import (prev.fetchFromGitHub {
          owner = "teutat3s";
          repo = "nixpkgs";
          rev = "ve889ab5";
          hash = "sha256-6bIhM+ZtAyp1omKACtKIXXJecE/SAPVL4Demp1mb15w=";
        }) {inherit system;})
        .vagrant;
    };

File: pkgs/development/tools/vagrant/default.nix [hunk 1]
```diff
@@ -5,9 +5,9 @@
 let
   # NOTE: bumping the version and updating the hash is insufficient;
   # you must use bundix to generate a new gemset.nix in the Vagrant source.
-  version = "2.3.4";
+  version = "2.3.7";
   url = "https://github.com/hashicorp/vagrant/archive/v${version}.tar.gz";
-  sha256 = "sha256-Q+sUYcbc/SOgw4ZXDmwqh24G0jiLvA8fDJyZ45OqLw8=";
+  sha256 = "sha256-+oqWMZqnuf9fSpkbd8vzf1SVSdhHN2JLzr76jyAEv0U=";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vagrant/default.nix:10]:
```suggestion
  hash = "sha256-+oqWMZqnuf9fSpkbd8vzf1SVSdhHN2JLzr76jyAEv0U=";
```
Nit

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/tools/vagrant/default.nix (hunk 1 above)
Comment (teutat3s) [reply pkgs/development/tools/vagrant/default.nix]: Addressed in https://github.com/NixOS/nixpkgs/pull/240996/commits/0ddd921ebd0c3045382e0539c328d3ebf0bb4e5a

**Comment by teutat3s** [comment]
😄 aaand yesterday, they released a new version: https://github.com/hashicorp/vagrant/blob/v2.4.0/CHANGELOG.md

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by teutat3s** [comment]
@SuperSandro2000 Is there anything missing here for a merge? :innocent:

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 240996` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vagrant</li>
  </ul>
</details>

Runs fine, LGTM. thanks!

**Pull request merged**